### DISCUSSION TITLE
No bounds on calendar means web spiders crawl the links relentlessly

### DISCUSSION THREAD

**Discussion by elibird** [solspace/craft-calendar#180] [Feature Requests]
I am maintaining a Craft 3 site w/ the calendar plugin, and noticed that Craft's /storage/runtime/cache folder was filling up with tens of thousands of files and using tens of gigabytes of space.

I tracked the issue down by examining the webserver's access logs and found various spiders crawling the "next" and "last" links, resulting in dates both very far in the future and past.

I modified the twig templates (month.html, day.html, week.html) to include something like the following:

```
{% set month = craft.calendar.month({
    date: targetDate,
    calendar: calendarHandle,
}) %}

{% set the_diff = now.diff(month.nextDate)|date('%r%a') %}

<!-- CALENDAR NAVIGATION - MONTHS -->
<div class="calendar-nav mb-3">

    {% if the_diff > -365 %}
        <div class="btn-prev">
            <a class="btn btn-outline-secondary"
href="{{ siteUrl }}{{ calendarBasePath }}{{ month.previousDate.format('Y/m') }}{{ urlParamsString }}">
                <span class="fas fa-arrow-left"></span><span class="d-none d-sm-inline"> {{ month.previousDate.format('F') }}</span>
            </a>
        </div>
    {% else %}
        <div><!-- no link for spiders, but add an empty div to not break the layout --></div>
    {% endif %}
```

So, this is more of a feature request, but it would be nice if I could go into the general settings of the plugin and set bounds on the links like I've done in the template, so that at a certain length of time either forwards or backwards, the "next" and "last" links no longer appear.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.7.57
- Calendar Version: 3.3.16
- Calendar Edition: Pro
- Fresh Install or Upgrade: Fresh

**Comment by kjmartens** [comment]
Sorry for the delay @elibird.

Thanks for this suggestion. I'll make a note of the bounds setting idea for future consideration, and we'll also consider adding the workaround to the demo templates and documentation. 🙂

**Reply by elibird** [reply]
That would be great! It'll be sure to save someone a headache down the road.

**Comment by kjmartens** [comment]
We've updated our documentation as well as demo template examples in Calendar 4.0.2+ to include your suggested template-level approach. Thank you! 🙂

**Reply by andrewmenich** [reply]
Just dropping a link to the commit for anyone else that comes across this issue on older projects https://github.com/solspace/craft-calendar/commit/5926d2fc7c6379bb96d674014aa8789bd941e1c4

**Comment by ahockemeyer** [comment]
Has this plugin be fixed? Our cache is out of control.
![image (6)](https://github.com/user-attachments/assets/e16ea15c-cbfb-49b5-93b9-da75913431ce)

**Reply by kjmartens** [reply]
I'm not so sure we can build a solution that would stop this and work for everyone (for example, what if someone wants users to be able to click past 1 year, etc). We recommend the template-level approach above. 🙂
